date_range/models/date_range_type.py

```diff
@@ -42,3 +71,42 @@ def _check_company_id(self):
                         _('You cannot change the company, as this '
                           'Date Range Type is  assigned to Date Range '
                           '(%s).') % (rec.date_range_ids.name_get()[0][1]))
+
+    @api.depends("date_range_ids")
+    def _compute_date_ranges_exist(self):
+        for dr_type in self:
+            dr_type.date_ranges_exist = bool(dr_type.date_range_ids)
+
+    @api.onchange("name_expr")
+    def onchange_name_expr(self):
+        """Wipe the prefix if an expression is entered.
+
+        The reverse is not implemented because we don't want to wipe the
+        users' painstakingly crafted expressions by accident.
+        """
+        if self.name_expr and self.name_prefix:
+            self.name_prefix = False
+
+    def test_name_expr(self):
+        """Button on the form to preview newly generated names"""
```

Comment:
@StefanRijnhart I would be nice to have a computed field `name_preview` displayed next to the `name_expr` to always display the preview without having to push a button. Moreover, if the expression is wrong, the UserError will prevent to store the value at save time. If you make this new field `store=True` you'll ensure that this implicit validation will always be applied even if `name_expr` is modified into the code.

Reply:
Done now. Screenshots updated in top comment. Thanks for suggesting!